services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusMigrationService.kt

```diff
@@ -28,19 +31,48 @@ class HakemusMigrationService(
     fun migrateOneHanke(hankeId: Int) {
         val hanke = hankeRepository.getReferenceById(hankeId)
 
-        hanke.hakemukset.singleOrNull()?.let {
-            val founder = createFounderKayttaja(hanke.id, it.applicationData)
-            val otherKayttajat = createOtherKayttajat(it.applicationData, founder, hanke.id)
-            createYhteystiedot(it, (otherKayttajat + founder).filterNotNull())
-            it.applicationData = clearCustomers(it.applicationData)
-        }
+        hanke.hakemukset.singleOrNull()?.let { hakemus: ApplicationEntity ->
+            logger.info { "Migrating hakemus. ${hakemus.logString()}, ${hanke.logString()}" }
+
+            val existingKayttajat = hankekayttajaRepository.findByHankeId(hanke.id)
+            logger.info {
+                if (existingKayttajat.isNotEmpty()) {
+                    "Hanke already has users. ${existingKayttajat.size} users with IDs: " +
+                        existingKayttajat.map { it.id }.joinToString()
+                } else {
+                    "Hanke does not have users."
+                }
+            }
+            val existingKayttajaEmails = existingKayttajat.map { it.sahkoposti }.toMutableSet()
+
+            val founder =
+                createFounderKayttaja(hanke.id, hakemus.applicationData, existingKayttajaEmails)
+            if (founder != null) {
+                logger.info { "Created founder with id ${founder.id}" }
+                existingKayttajaEmails.add(founder.sahkoposti)
+            } else {
+                logger.info { "No founder found in hakemus data." }
+            }
+            val otherKayttajat =
+                createOtherKayttajat(hakemus.applicationData, existingKayttajaEmails, hanke.id)
```

Comment:
It might be better to make `createFounderKayttaja` and `createOtherKayttajat` more uniform by setting the parameters in the same order: `hanke.id, hakemus.applicationData, existingKayttajaEmails`.